packages/react-components/src/components/DevicesButton.tsx

```diff
@@ -172,12 +172,28 @@ const generateDefaultMenuProps = (
 
   const defaultMenuProps: IContextualMenuProps = {
     items: [],
-
     styles: props.styles?.menuStyles,
-
-    // Confine the menu to the parents bounds.
-    // More info: https://github.com/microsoft/fluentui/issues/18835
-    calloutProps: { styles: { root: { maxWidth: '95%' } } }
+    calloutProps: {
+      styles: {
+        root: {
+          // Confine the menu to the parents bounds.
+          // More info: https://github.com/microsoft/fluentui/issues/18835
+          maxWidth: '100%'
+        }
+      },
+      // Disable dismiss on resize to work around a couple Fluent UI bugs
+      // - The Callout is dismissed whenever *any child of window (inclusive)* is resized. In practice, this
+      //   happens when we change the VideoGallery layout, or even when the video stream element is internally resized
+      //   by the headless SDK.
+      // - There is a `preventDismissOnEvent` prop that we could theoretically use to only dismiss when the target of
+      //   of the 'resize' event is the window itself. But experimentation shows that setting that prop doesn't
+      //   deterministically avoid dismissal.
+      //
+      // A side effect of this workaround is that the context menu stays open when window is resized, and may
+      // get detached from original target visually. That bug is preferable to the bug when this value is not set -
+      // The Callout (freuently) gets dismissed automatically.
```

Comment:
I mean there's no ambiguity to what this word meant so very pointless nit: `freuently` -> `frequently`